api/clamav_scanner/clamav.py

```diff
@@ -0,0 +1,205 @@
+from .common import AV_DEFINITION_FILE_SUFFIXES
+from .common import AV_SIGNATURE_OK
+from .common import AV_SIGNATURE_UNKNOWN
+from .common import AV_STATUS_CLEAN
+from .common import AV_STATUS_INFECTED
+from .common import CLAMAVLIB_PATH
+from .common import CLAMSCAN_PATH
+from .common import FRESHCLAM_PATH
+from .common import create_dir
+
+from boto3wrapper.wrapper import get_session
+
+
+RE_SEARCH_DIR = r"SEARCH_DIR\(\"=([A-z0-9\/\-_]*)\"\)"
+
+
+def current_library_search_path():
+    ld_verbose = subprocess.check_output(["ld", "--verbose"]).decode("utf-8")
+    rd_ld = re.compile(RE_SEARCH_DIR)
+    return rd_ld.findall(ld_verbose)
+
+
+def update_defs_from_s3(s3_client, bucket, prefix):
+    create_dir(AV_DEFINITION_PATH)
+    to_download = {}
+    for file_prefix in AV_DEFINITION_FILE_PREFIXES:
+        s3_best_time = None
+        for file_suffix in AV_DEFINITION_FILE_SUFFIXES:
+            filename = file_prefix + "." + file_suffix
+            s3_path = os.path.join(AV_DEFINITION_S3_PREFIX, filename)
+            local_path = os.path.join(AV_DEFINITION_PATH, filename)
+            s3_md5 = md5_from_s3_tags(s3_client, bucket, s3_path)
+            s3_time = time_from_s3(s3_client, bucket, s3_path)
+
+            if s3_best_time is not None and s3_time < s3_best_time:
+                print("Not downloading older file in series: %s" % filename)
+                continue
+            else:
+                s3_best_time = s3_time
+
+            if os.path.exists(local_path) and md5_from_file(local_path) == s3_md5:
+                print("Not downloading %s because local md5 matches s3." % filename)
+                continue
+            if s3_md5:
+                to_download[file_prefix] = {
+                    "s3_path": s3_path,
+                    "local_path": local_path,
+                }
+    return to_download
+
+
+def upload_defs_to_s3(s3_client, bucket, prefix, local_path):
+    for file_prefix in AV_DEFINITION_FILE_PREFIXES:
+        for file_suffix in AV_DEFINITION_FILE_SUFFIXES:
+            filename = file_prefix + "." + file_suffix
+            local_file_path = os.path.join(local_path, filename)
+            print("search for file %s" % local_file_path)
+            if os.path.exists(local_file_path):
+                local_file_md5 = md5_from_file(local_file_path)
+                if local_file_md5 != md5_from_s3_tags(
+                    s3_client, bucket, os.path.join(prefix, filename)
+                ):
+                    print(
+                        "Uploading %s to s3://%s"
+                        % (local_file_path, os.path.join(bucket, prefix, filename))
+                    )
+                    if os.environ.get("AWS_LOCALSTACK", False):
+                        s3 = get_session().resource(
+                            "s3", endpoint_url="http://localstack:4566"
+                        )
+                    else:
+                        s3 = get_session().resource("s3")
```

Comment:
I've been thinking about this pattern because of how I've seen it used in other Lambdas.  Would it be possible to do this test once for a constant and then use that to set the `endpoint_url` in all cases?

Something like so if it works:
```python
# Init vars
IS_LOCALSTACK = os.environ.get("AWS_LOCALSTACK", False)
AWS_ENDPOINT_URL = "http://localstack:4566" if IS_LOCALSTACK else None

# And then we can replace the if/else blocks with
 s3 = get_session().resource("s3", endpoint_url=AWS_ENDPOINT_URL)
```

Reply:
much cleaner